triton/stream.py

```diff
@@ -234,7 +236,12 @@ def _select_shard_ids(self, shard_nums):
         return shard_ids
 
     def put(self, **kwargs):
-        data = msgpack.packb(kwargs)
+        try:
+            data = msgpack.packb(kwargs)
+        except TypeError:
+            # If we fail to serialize our context, we can try again with an
+            # enhanced packer (it's slower though)
```

Comment:
a lot slower?   Would it be preferable to have the user cast their data into floats, or avoid using decimals all together if they expect or need high velocity writes?